### DISCUSSION TITLE
Is it possible to resize an existing table?

### DISCUSSION THREAD

**Discussion by NeilMacMullen** [ShapeCrawler/ShapeCrawler#783] [Q&A] [answered]
First of all, nice library!  I'm trying to use this to automate generation of powerpoint reports from data in my tool https://github.com/NeilMacMullen/kusto-loco/wiki/LokqlDX

One of the things I'd like to do is update an existing named table in a pptx template file.  The size of the data may change over time so I'd like to be able to resize the  table when inserting content....

It seems it's _almost_ possible to do this but unless I'm mistaken there's no way to add a column ?

``` CSharp
void ResizeTable(ITable existingTable,int requiredRows,int requiredColums)
{
   while (existingTable.Rows.Count > requiredRows) existingTable.Rows.RemoveAt(0);
   while (existingTable.Rows.Count < requiredRows) existingTable.Rows.Add();
   while (existingTable.Columns.Count > requiredColumns) existingTable.RemoveColumnAt(0);
   while (existingTable.Columns.Count> requiredColumns) .../* what goes here? */ ;
}
```

Am I missing something?  Obviously I can add a _new_ table with an arbitrary size but I'd like to insert cells into the the existing item in order to maintain placement within the presentation.  Maybe there's an alternate approach that relies upon creating a new table inside some kind of named container?

Thanks in advance for any tips you can offer...

**Comment by AdamShakhabov** [comment]
Hello @NeilMacMullen. Thank you for your feedback👍

I've added new methods to add column. You can test them using the latest beta version:

```Shell
dotnet add package ShapeCrawler --version 0.56.1-beta.202411231758 --source https://www.myget.org/F/shape/api/v3/index.json
```

```C#
var pres = new Presentation("pres.pptx");
var table = pres.Slide(1).Table("Table 1");

table.AddColumn();
table.InsertColumnAfter(1);
```

**Reply by NeilMacMullen** [reply]
Wow - thanks for such a quick patch! I have the code working now using 

```CSharp
while (table.Columns.Count < requiredColumns)
   table.InsertColumnAfter(table.Columns.Count);
```

I wasn't able to get it working using `table.AddColumn()` .  In that case the columns were added but it was not possible to set the content of the new cells.  I'd assumed AddColumn would simply call through to InsertColumnAfter(this.Columns.Count) but looking at the code you've added it behaves subtly differently. Is your intention that AddColumn should only be used for the very first column in an otherwise empty table?  E.g.

```CSharp
while (table.Columns.Count < requiredColumns)
{
  if (table.Colums.Count ==0) 
    table.AddColumn();
  else
    table.InsertColumnAfter(existingTable.Columns.Count);
}
```

**Reply by AdamShakhabov** [reply, marked as answer]
You are right: the implementation of `AddColumn()` should be `this.InsertColumnAfter(this.Columns.Count)`. It has been changed.

The content updating bug was also fixed. You can check it with the latest beta version 0.56.1-beta.202411232040

**Reply by NeilMacMullen** [reply]
That's great - thank you! :-)
